cmd/crc/cmd/daemon.go

```diff
@@ -199,18 +197,46 @@ func run(configuration *types.Configuration) error {
 		}
 	}()
 
-	go func() {
-		mux := http.NewServeMux()
-		mux.Handle(types.ConnectPath, vn.Mux())
-		s := &http.Server{
-			Handler:      mux,
-			ReadTimeout:  10 * time.Second,
-			WriteTimeout: 10 * time.Second,
-		}
-		if err := s.Serve(vsockListener); err != nil {
-			errCh <- errors.Wrap(err, "virtualnetwork http.Serve failed")
+	if runtime.GOOS == "darwin" {
+		go func() {
+			_ = os.Remove(constants.UnixgramSocketPath)
+			conn, err := transport.ListenUnixgram(fmt.Sprintf("unixgram://%v", constants.UnixgramSocketPath))
+			if err != nil {
+				errCh <- errors.Wrap(err, "failed to listen unixgram")
+			}
+			fmt.Println("Listening on:", constants.UnixgramSocketPath)
+			vfkitConn, err := transport.AcceptVfkit(conn)
+			if err != nil {
+				errCh <- errors.Wrap(err, "failed to accept vfkit connection")
+			}
+			ctx := context.Background()
+			err = vn.AcceptVfkit(ctx, vfkitConn)
+			if err != nil {
+				// Don't return an error if the connection is closed
+				if !errors.Is(err, net.ErrClosed) {
+					errCh <- errors.Wrap(err, "failed to accept vfkit connection")
+				}
+				return
+			}
```

Comment:
Imo this belongs more in `daemon_darwin.go` than here, this at least should go in a separate function as it is quite long.